lib/Driver/ToolChains/Flang.cpp

```diff
@@ -774,9 +775,7 @@ void FlangFrontend::ConstructJob(Compilation &C, const JobAction &JA,
 
   UpperCmdArgs.push_back("-output");
   if (Args.hasArg(options::OPT_E)) {
-    if (Arg *A = Args.getLastArg(options::OPT_o)) {
-      UpperCmdArgs.push_back(Args.MakeArgString(A->getValue()));
-    }
+    UpperCmdArgs.push_back(llvm::sys::path::filename(OutFile));
```

Comment:
`OutFile` is NOT the "output file." It is the LLVM IR file that goes into the optimizer.